### ISSUE TITLE
rancher-compose rm doesn't block until the services are deleted from the DB

### ISSUE THREAD

**Issue by juliangamble** [rancher/rancher#4736]
**Rancher Version:**
Output:

    Rancher	v1.0.1
    Cattle	v0.159.7
    User Interface	v1.0.5
    Rancher Compose	v0.7.4

**Docker Version:**

Client:`docker-compose version: 1.5.1`

**OS and where are the hosts located? (cloud, bare metal, etc):**
Redhat, AWS Cloud. 

**Setup Details: (single node vs. single node, internal DB vs. external DB)**

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle. 

**Steps to Reproduce:**
`rancher-compose -p mystack rm --force -v`

**Results:** 
    22:55:22.118 time="2016-05-07T22:55:22+10:00" level=info msg="Project [myproject]: Deleting project " 
    22:55:22.119 time="2016-05-07T22:55:22+10:00" level=info msg="[0/3] [service-a]: Deleting " 
    22:55:22.119 time="2016-05-07T22:55:22+10:00" level=info msg="[0/3] [service-primary]: Deleting " 
    22:55:22.119 time="2016-05-07T22:55:22+10:00" level=info msg="[0/3] [sidekick]: Deleting " 
    22:55:22.127 time="2016-05-07T22:55:22+10:00" level=info msg="[0/3] [service-primary]: Deleted " 
    22:55:22.128 time="2016-05-07T22:55:22+10:00" level=info msg="[0/3] [sidekick]: Deleted " 
    22:55:22.139 time="2016-05-07T22:55:22+10:00" level=info msg="[0/3] [service-a]: Deleted " 
    22:55:22.140 Clearing requested, pausing to allow command to complete...

The stack is not deleted, but the results say that it is. (Of course it could be requested asynchronously, but the output suggests that this is not the case). 

Currently the online rancher docs recommend working around this by doing a `sleep`.   :0

https://github.com/rancher/rancher/issues/3622
(note this issue is closed, but the software remains defective - I suggest you leave this issue open). 

**Expected:**
The option to do a synchronous `docker-compose rm`.

**Comment by deniseschannon** [comment]
> The stack is not deleted, but the results say that it is. (Of course it could be requested asynchronously, but the output suggests that this is not the case).

We do not have an option to remove the stack. There is an error in the logs where it's claiming to remove the stack when we don't even make the call to remove the stack in Rancher.

https://github.com/rancher/rancher/issues/4787

As for the services being deleted, we have made the API call to get the services deleted, but we don't do any checks to see if the service has been removed and purged from Rancher. 

I'll leave this open as an enhancement to wait to remove services, but the workaround is to add a `sleep`.
